src/connect4/core.rs

```diff
@@ -366,26 +383,80 @@ impl Board {
     pub fn insert(&mut self, position: i32, team: i32, color: MyColor) -> bool {
         self.columns[position as usize].insert(team, color)
     }
+    
+    pub fn reset(&mut self) {
+        for column in &mut self.columns {
+            column.reset();
+        }
+    }
+}
 
+pub struct TurnIndicator {
+    team: i32,
 }
 
+impl TurnIndicator {
+    pub fn new() -> Self {
+        TurnIndicator {
+            team: 0,
+        }
+    }
+
+    fn draw(&self, ctx: &mut Context) -> GameResult<()> {
+        let mut text: graphics::Text;
+        if self.team == 0 {
+            text = graphics::Text::new(("Paused", graphics::Font::default(), TURN_INDICATOR_FONT_SIZE as f32));
+        } else {
+            text = graphics::Text::new((format!("Player {}'s turn", self.team), graphics::Font::default(), TURN_INDICATOR_FONT_SIZE as f32));
+        }
```

Comment:
Below is the dynamic sizing and position of the textbox according to size of the drawn text.
`&text.dimensions(ctx)` returns a tuple containing width and height.
`y pos` doesn't change much because text grows length-wise.